airsonic-main/src/main/java/org/airsonic/player/service/StatusService.java

```diff
@@ -43,35 +45,27 @@
     @Autowired
     private MediaFileService mediaFileService;
 
-    private final List<TransferStatus> streamStatuses = new ArrayList<TransferStatus>();
-    private final List<TransferStatus> downloadStatuses = new ArrayList<TransferStatus>();
-    private final List<TransferStatus> uploadStatuses = new ArrayList<TransferStatus>();
-    private final List<PlayStatus> remotePlays = new ArrayList<PlayStatus>();
+    private final List<TransferStatus> streamStatuses = Collections.synchronizedList(new ArrayList<>());
+    private final List<TransferStatus> downloadStatuses = Collections.synchronizedList(new ArrayList<>());
+    private final List<TransferStatus> uploadStatuses = Collections.synchronizedList(new ArrayList<>());
+    private final List<PlayStatus> remotePlays = Collections.synchronizedList(new ArrayList<>());
 
     // Maps from player ID to latest inactive stream status.
-    private final Map<Integer, TransferStatus> inactiveStreamStatuses = new LinkedHashMap<Integer, TransferStatus>();
-
-    public synchronized TransferStatus createStreamStatus(Player player) {
-        // Reuse existing status, if possible.
-        TransferStatus status = inactiveStreamStatuses.get(player.getId());
-        if (status != null) {
-            status.setActive(true);
-        } else {
-            status = createStatus(player, streamStatuses);
-        }
-        return status;
```

Comment:
Sorry not quite following, why don't we need this check for existing `TransferStatus` for the given player?

Reply:
This was a while back so I'm trying to refresh my memory, but I think the more pertinent question is why do you need this check in the first place?

Are you trying to maintain one stream status per player at all times?

That's already not going to happen: if your player hasn't been inactive, every stream transfer status creation will create a new transfer status and add it to the active list (because it won't find anything in the inactive map, and no one is checking the active list), leading to multiple statuses per player in the active list. That's just inconsistent behavior, gated on when a stream status for a player turns inactive for the first time.

Additionally think of the concurrency scenarios. A transfer status being used for a transfer of a file it marks, suddenly gets interrupted by another thread for another file. Now two threads are vying for the transfer status, and while the byte counts may be fine because of the atomic byte counts in there, the file path is certainly not accurate (the status's path will only reflect whatever the latest thread's file path is, even though the earlier thread is still using it for its own file).

I think this way was just cleaner. You can keep a map around for display of older inactive transfers, but always create a new one.